torch/csrc/jit/backends/backend_init.cpp

```diff
@@ -1,11 +1,19 @@
 #include <torch/csrc/jit/backends/backend_init.h>
-#include <torch/csrc/jit/backends/test_backend.h>
+#include <torch/csrc/jit/backends/backend_detail.h>
 
 namespace torch {
 namespace jit {
 
 void initJitBackendBindings(PyObject* module) {
-  initTestBackendBindings(module);
+  // Create Python bindings for all registered backends.
+  for (auto& p : detail::getRegisteredLoweringFunctions()) {
+    auto m = py::handle(module).cast<py::module>();
+    std::string binding_name("_jit_to_");
+    binding_name.append(p.first);
+    // If there is a registered backend called "example_backend", this generates
+    // a binding called torch._C._jit_to_example_backend.
+    m.def(binding_name.c_str(), p.second);
+  }
```

Comment:
This is going to break if someone's registering a backend later than static initialization. In the past we've followed a few patterns to get around this issue:

1) Have the initial registration do this (tricky with Python bindings because that code necessarily has to live in a separate `.so`), so the more viable alternatives are:
2) Build a "listener" infrastructure, basically a callback that gets installed that is called on registration (regardless of when in program initialization/runtime it happens) that does something like this. The listener needs to fire on all previously-registered objects at installation time as well.
3) Dynamically generate these binding when needed and cache those results away. This is how custom ops and classes work

Reply:
This is a good point. I think #3 will also help separate the Python and C++ code better because we won't have to include pybind stuff in the `backend.h` and `backend_detail.h` headers at all.

Reply:
Updated, what do you think now?

Reply:
Okay, I think I implemented #2, please take a look.